controllers/status/status.go

```diff
@@ -210,7 +216,135 @@ func RemoveComponentCondition(conditions *[]conditionsv1.Condition, component st
+}
+
+// +k8s:deepcopy-gen=true
+type CodeFlareStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DashboardStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type WorkbenchesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ModelMeshServingStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DataSciencePipelinesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KserveStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KueueStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type RayStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrustyAIStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
 type ModelRegistryStatus struct {
 	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
+	ComponentStatus     `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrainingOperatorStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentsStatus struct {
+	CodeFlare            *CodeFlareStatus            `json:"codeflare,omitempty"`
+	Dashboard            *DashboardStatus            `json:"dashboard,omitempty"`
+	DataSciencePipelines *DataSciencePipelinesStatus `json:"datasciencepipelines,omitempty"`
+	ModelMeshServing     *ModelMeshServingStatus     `json:"modelmeshserving,omitempty"`
+	ModelRegistry        *ModelRegistryStatus        `json:"modelregistry,omitempty"`
+	Kserve               *KserveStatus               `json:"kserve,omitempty"`
+	Kueue                *KueueStatus                `json:"kueue,omitempty"`
+	Ray                  *RayStatus                  `json:"ray,omitempty"`
+	TrustyAI             *TrustyAIStatus             `json:"trustyai,omitempty"`
+	TrainingOperator     *TrainingOperatorStatus     `json:"trainingoperator,omitempty"`
+	Workbenches          *WorkbenchesStatus          `json:"workbenches,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ReleaseFileMeta struct {
+	Releases []ComponentReleaseStatusMeta `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatusMeta struct {
+	Name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+	RepoURL string `yaml:"repoURL,omitempty"`
+}
+
+// GetReleaseVersion reads odh_metadata.yaml file and parses release information.
+// If version is not set or set to "", return empty {}.
+func GetReleaseVersion(defaultManifestPath string, componentName string) ComponentStatus {
+	var componentVersion semver.Version
+	var releaseInfo ReleaseFileMeta
+	var releaseStatus ComponentReleaseStatus
+	componentReleaseStatus := make([]ComponentReleaseStatus, 0)
+
+	yamlData, err := os.ReadFile(filepath.Join(defaultManifestPath, componentName, "odh_metadata.yaml"))
+	if err != nil {
```

Comment:
I think we need to distinguish between a file not present from an error, in this case we are swallowing it and I'm not sure that's the right approach as it could hide some other errors.